src/ConfigDefinition.php

```diff
@@ -2,22 +2,60 @@
 
 declare(strict_types=1);
 
-namespace MyComponent;
+namespace Keboola\FtpExtractor;
 
 use Keboola\Component\Config\BaseConfigDefinition;
 use Symfony\Component\Config\Definition\Builder\ArrayNodeDefinition;
 
 class ConfigDefinition extends BaseConfigDefinition
 {
+    public const CONNECTION_TYPE_FTP = 'FTP';
+    public const CONNECTION_TYPE_SSL_IMPLICIT = 'FTPS';
+    public const CONNECTION_TYPE_SFTP = 'SFTP';
+
     protected function getParametersDefinition(): ArrayNodeDefinition
     {
         $parametersNode = parent::getParametersDefinition();
         // @formatter:off
         /** @noinspection NullPointerExceptionInspection */
         $parametersNode
             ->children()
-                ->scalarNode('foo')
-                    ->defaultValue('baz')
+                ->scalarNode('host')
+                    ->isRequired()
+                    ->cannotBeEmpty()
+                ->end()
+                ->scalarNode('username')
+                    ->isRequired()
+                    ->cannotBeEmpty()
+                ->end()
+                ->scalarNode('password')
+                    ->isRequired()
+                    ->cannotBeEmpty()
+                ->end()
+                ->scalarNode('path')
+                    ->isRequired()
+                    ->cannotBeEmpty()
+                ->end()
+                ->booleanNode('onlyNewFiles')
+                    ->defaultFalse()
+                ->end()
+                ->booleanNode('wildcard')
+                    ->defaultFalse()
+                ->end()
+                ->integerNode('port')
+                    ->min(1)->max(65535)
+                ->end()
+                ->scalarNode('connectionType')
+                    ->isRequired()
+                    ->validate()->ifNotInArray([
+                               self::CONNECTION_TYPE_FTP,
+                               self::CONNECTION_TYPE_SSL_IMPLICIT,
```

Comment:
Nevím, tak se ptám a je to spíš pro-forma, ale tohle je FTPS (tj. FTP over SSL), takže je to SSL spíš explicitní, než implicitní, ne?

Reply:
http://www.serv-u.com/kb/1088/implicit-vs-explicit-ssl
Implicitní = komunikace probíhá rovnou po SSL
Eplicitní = komunikace se naváže na FTP(21) a jako první příkaz se pošle přepnutí na SSL.

PHP nativní fce podporují pouze Explicit (což jsem si splet a konstantu špatně pojmenoval). 
https://stackoverflow.com/questions/6589730/ftp-ssl-connect-with-implicit-ftp-over-tls

Původní extractor měl 3 možnosti (FTPS, FTPS_EXPLICIT, FTPS_IMPLICIT), kde FTPS jako takové "existovat nemůže". Buďto je to explicitní nebo implicitní. Ještě se do toho původního projektu podívám. Nicméně možnost EX/IMPLICIT používají jenom 2 projekty a s @Halama  jsme se dohodli, že dál bude prostě jedna možnost FTPS.